dallinger/recruiters.py

```diff
@@ -564,7 +572,7 @@ def parse_spec(self):
             recruiters.append((name, count))
         return recruiters
 
-    def pick_recruiter(self):
+    def pick_recruiter(self, n=1):
```

Comment:
Doc string could use an update, especially since it's a little counter-intuitive to ask for 3 (the _total_ number you want, across all recruitment types) and get back 2 (the number for the next type only). This initially confused me when looking at the tests.

Reply:
Docstring seems wise here, though I think returning the number available to be recruited for each recruiter you iterate over is the most reasonable way to do this. Now I'm thinking this should maybe be an iterator that `yield`s `recruiter, count` tuples, that would clarify the meaning of `n` a bit as well.

Reply:
I think the iterator would help. Re-reading this method I also felt my brain wilt on this:
```
        counts = dict(
            session.query(
                Recruitment.recruiter_id,
                func.count(Recruitment.id)
            ).group_by(Recruitment.recruiter_id).all()
        )
```
Could that either be moved to a separate method with a docstring showing the returned data structure, or else just a little inline comment with that structure? A method would be better because simply naming it would help.